src/plus/integrations/authentication/github.ts

```diff
@@ -1,16 +1,47 @@
 import type { AuthenticationSession, Disposable, QuickInputButton } from 'vscode';
-import { env, ThemeIcon, Uri, window } from 'vscode';
-import type {
-	IntegrationAuthenticationProvider,
-	IntegrationAuthenticationSessionDescriptor,
+import { authentication, env, ThemeIcon, Uri, window } from 'vscode';
+import { wrapForForcedInsecureSSL } from '@env/fetch';
+import { HostingIntegrationId, SelfHostedIntegrationId } from '../providers/models';
+import type { IntegrationAuthenticationSessionDescriptor } from './integrationAuthentication';
+import {
+	CloudIntegrationAuthenticationProvider,
+	LocalIntegrationAuthenticationProvider,
 } from './integrationAuthentication';
 
-export class GitHubEnterpriseAuthenticationProvider implements IntegrationAuthenticationProvider {
-	getSessionId(descriptor?: IntegrationAuthenticationSessionDescriptor): string {
-		return descriptor?.domain ?? '';
+export class GitHubAuthenticationProvider extends CloudIntegrationAuthenticationProvider<HostingIntegrationId.GitHub> {
+	protected override get authProviderId(): HostingIntegrationId.GitHub {
+		return HostingIntegrationId.GitHub;
 	}
 
-	async createSession(
+	override async getBuiltInExistingSession(
+		descriptor?: IntegrationAuthenticationSessionDescriptor,
+		options?: { authorizeIfNeeded?: boolean; createIfNeeded?: boolean; forceNewSession?: boolean },
```

Comment:
Because this logic is serving only to get an existing session, not to create one, we should not need any of these options. On the `authentication.getSession` call, we can set `silent: true` and not send any other options.

Reply:
ok

Reply:
done

![image](https://github.com/gitkraken/vscode-gitlens/assets/83944326/37fe8bf1-9f40-49cd-9ff4-21c72f500f86)